src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -750,13 +753,26 @@ public static String getNodeHostname(@Nullable EnvVars envVars, @Nullable Comput
                 if (DatadogUtilities.isValidHostname(computerHostName)) {
                     return computerHostName;
                 }
+
+                Node node = computer.getNode();
+                if (node != null) {
+                    FilePath rootPath = node.getRootPath();
+                    if (rootPath != null) {
+                        ShellCommandCallable hostnameCommand = new ShellCommandCallable(
+                                Collections.emptyMap(), HOSTNAME_CMD_TIMEOUT_MILLIS, "hostname", "-f");
```

Comment:
What do you think about having the same approach we use here, maybe creating a helper method? https://github.com/jenkinsci/datadog-plugin/blob/9e31a06793655958e1cc377bcab97cc9794d5495/src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java#L534

Reply:
Good point, I agree that it is better to check the OS explicitly. Although reusing the same approach we use for the controller node wouldn't be accurate, as we're accessing a remote worker which may in theory be running a different OS than the controller.
I used a remote callable instead - the same approach that is used in Jenkins internally to determine whether a node is running a Unix-flavoured OS.

Reply:
Mhm I see, that makes sense. Do you mind adding some logging so we know when this command fails?

Reply:
The callable will throw an exception if the command fails. Updated the catch block to print exception along with the computer node name